### PR TITLE
Remove broker configuration

### PR DESCRIPTION
**Pull request by pjvv**

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -55,14 +55,7 @@
...
   config.events = {
-    enabled: false,
-    broker_host: 'localhost',
-    broker_port: '5672',
-    broker_username: 'guest',
-    broker_password: 'guest',
-    exchange_name: 'aker.events',
-    warehouse_queue_name: 'aker.events.warehouse',
-    notification_queue_name: 'aker.events.notifications'
+    enabled: false
```
Comment (NikolausC) [inline config/environments/test.rb]: Does this still need to be kept in a hash now that it's only one item?

File: Gemfile [hunk 2]
```diff
@@ -8,111 +8,80 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
-# REST on Rails. Wrap your RESTful web app with Ruby classes and work with them like Active Record
-#   models. Read more: https://github.com/rails/activeresource
+# All the gems not in a group will always be installed:
```
Comment (NikolausC) [inline Gemfile]: I like the tidy Gemfile

File: README.md [hunk 3]
```diff
@@ -3,18 +3,104 @@
...
+To create the exchanges, queues and usernames etc. use the GitLab repo: [aker-environments](https://gitlab.internal.sanger.ac.uk/aker/aker-environments)
+
 # Testing
 To run ALL the tests for this project: `bundle exec rake`
 
 To run just the rspec tests: `bundle exec rspec`
 
-Note that if any changes are made in the `app/client`,`app/assets` or `app/javascript` directories, any files in the `public/assets`,`public/packs` or `public/packs-test`  directory must be removed, and webpack must be run with the `webpack.production.js` configuration file:
+# Misc.
+## Assets
```
Comment (NikolausC) [inline README.md]: Don't we only have to do this for set shaper? 😨

File: README.md [hunk 4]
```diff
@@ -3,18 +3,104 @@
...
     RAILS_ENV=test bundle exec rake assets:precompile
-    RAILS_ENV=production bundle exec rake assets:precompile
\ No newline at end of file
+    RAILS_ENV=production bundle exec rake assets:precompile
+
+## Messages
+## Catalogue message
+The following message can be used to create a catalogue but currently process names have to be
+unique so adjust the message as required.
+```json
```
Comment (NikolausC) [inline README.md]: This could be replaced with the larger catalog (pending code-review)

File: config/initializers/broker_initializer.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'broker'
+
+if Rails.configuration.events[:enabled]
```
Comment (NikolausC) [inline config/initializers/broker_initializer.rb]: `Rails.configuration.events_enabled` if we ditch the hash

File: spec/models/work_order_spec.rb [hunk 6]
```diff
@@ -1,237 +1,248 @@
...
 
   def make_barcode
     @barcode_index += 1
     "AKER-#{@barcode_index}"
   end
 
-  def make_set(size=6)
+  def make_set(size = 6)
     uuid = make_uuid
     s = double(:set, uuid: uuid, id: uuid, meta: { 'size' => size })
```
Comment (NikolausC) [inline spec/models/work_order_spec.rb]: If we're using an implicit return, could we rename 's' to something more descriptive, like 'set'?

File: spec/lib/event_publisher_spec.rb [hunk 7]
```diff
@@ -119,13 +132,13 @@ def mock_connection(params)
         @unconfirmed_sets = Set.new([1])
       end
 
-      it 'raises exception if unconfirmed set is not empty' do
+      it 'raises excbrokertion if unconfirmed set is not empty' do
```
Comment (NikolausC) [inline spec/lib/event_publisher_spec.rb]: Typo

File: lib/broker.rb [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  def create_connection
+    !connected? && connect!
+  end
+
+  def publish(message)
+    create_connection unless connected?
+    Rails.logger.debug('Publishing message to broker')
+    @exchange.publish(message.generate_json, routing_key: EventMessage::ROUTING_KEY)
+    @channel.wait_for_confirms
+    raise 'There is an unconfirmed set.' if @channel.unconfirmed_set.count.positive?
```
Comment (NikolausC) [inline lib/broker.rb]: What's an unconfirmed set?

File: spec/lib/event_publisher_spec.rb [hunk 9]
```diff
@@ -3,114 +3,127 @@
 require 'rails_helper'
```
Comment (NikolausC) [inline spec/lib/event_publisher_spec.rb]: Might be worth renaming to broker_spec.rb

File: spec/models/work_order_spec.rb [hunk 10]
```diff
@@ -1,237 +1,248 @@
...
+               cost_code: cost_code,
+               id: id,
+               parent_id: parent_id,
+               subproject?: is_sub,
+               project?: is_proj,
+               node_uuid: make_uuid,
+               data_release_uuid: data_release_uuid)
     allow(StudyClient::Node).to receive(:find).with(n.id).and_return([n])
-    return n
+    n
```
Comment (NikolausC) [inline spec/models/work_order_spec.rb]: n -> node

File: Gemfile (hunk 2 above)
Comment (pjvv) [reply Gemfile]: 👍

File: README.md (hunk 3 above)
Comment (pjvv) [reply README.md]: Hmmm... I think now that it uses React we need to be aware of assets.

File: README.md (hunk 4 above)
Comment (pjvv) [reply README.md]: True, it would be helpful to keep this message up-to-date.

File: lib/broker.rb (hunk 8 above)
Comment (pjvv) [reply lib/broker.rb]: > Set of published message indexes that are currently unconfirmed

**Pull request merged**